src/components/nav-icon-item/styles.ts

```diff
@@ -0,0 +1,61 @@
+import { styled } from '@linaria/react'
+import { AnchorHTMLAttributes, ButtonHTMLAttributes } from 'react'
+
+type CommonNavIconItemProps = {
+  /**
+   * NOTE: data attribute to indicate the state of the nav item
+   * by default the styles is already applied with pseudo-class selector
+   * but to make it more flexible, consumer could use this attribute to
+   * change the styles
+   */
+  'data-state'?: 'focus' | 'hover' | 'active'
+}
+
+const baseStyles = `
+  display: inline-flex;
+  padding: var(--spacing-2, 8px);
+  justify-content: center;
+  align-items: center;
+  gap: var(--spacing-none, 0px);
+  border-radius: var(--corner-default, 0.25rem);
+  background: var(--white, #ffffff);
+  border: none;
+  outline: none;
+
+  /* change children color by default */
+  > * {
+    color: var(--neutral-400, #798da1) !important;
+  }
```

Comment:
**question:** Do we strictly need this right now? Ideally our icon components would just inherit the colour of their parent. I'm just wary of use forcing styles on descendants like this...

Same comment applies to the active styles further down.

Reply:
> Do we strictly need this right now? Ideally our icon components would just inherit the colour of their parent. I'm just wary of use forcing styles on descendants like this

I’ve thought about this previously; but when I try to add it to Storybook using `Icon` from Elements, the color won’t change since they have their own isolated color management. So, in the end, I just force-change the color, no matter what the children of `NavIconItem` have, what do you think?

Reply:
Ah, okay, so it is necessary at the moment. That's a shame.

**thought:** could we import the class name for the existing Icon component and target it here instead of `*`? If not, let's improve the comment you have so that we explain _why_ we overriding the colour of our descendants/children here (the current comment isn't very helpful because it's just describing what the CSS already communicates).

Reply:
> could we import the class name for the existing Icon component and target it here instead of *?

I'll try it first, and see how far it goes

Reply:
**note:** I think the wise way is to ensure the child component to have `color:inherit;` style

**thought:** the upcoming commit would import the `ElIcon` class name and have a comment to ensure the child of this component should have `inherit` style